src/cpp/detail/data_getters.hpp

```diff
@@ -191,11 +191,12 @@ struct StatisticsIterator<database::PublicationThroughputSample> final
 
     double get_value() const noexcept override
     {
-        auto aux_prev = *this - 1;
-        auto& prev = reinterpret_cast<decltype(*this)>(aux_prev);
-        double diff = sample().data - prev.sample().data;
+        auto aux_prev = static_cast<base_iterator_type>(*this);
+        --aux_prev;
+        StatisticsIterator<database::EntityDataSample> prev(aux_prev);
+        double diff = sample().data - prev.get_value();
         auto diff_time = get_timestamp() - prev.get_timestamp();
-        return diff / diff_time.count();
+        return diff * (1e+9 / diff_time.count());
```

Comment:
The finite differentiation is already done in origin:
https://github.com/eProsima/Fast-DDS/blob/5b8b9058c1e8c91f498f0f09dfaf724d65a4ff62/src/cpp/statistics/rtps/writer/StatisticsWriterImpl.cpp#L198-L200